### PR TITLE
Fixed two major issues with event scheduling

### PR DESCRIPTION
**Pull request by madeye-matt**
- Due to using notNull instead of null, when the collection exercise
  service started up it would schedule quartz events only for those
  events for which a message had already been sent
- Attempts to update the datetime for an existing event would fail on
  account of the scheduled job already existing.  However this bug was
  masked to some extent by the issue above (as many events were not
  scheduled, the first attempt to PUT a new time would succeed as
  there was no previous quartz event)

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/message/impl/CollectionExerciseEventInboundReceiver.java [hunk 1]
```diff
@@ -34,19 +34,33 @@
...
-
-            try {
-                this.collectionExerciseService.transitionCollectionExercise(collexId,
-                        CollectionExerciseDTO.CollectionExerciseEvent.GO_LIVE);
-            } catch (CTPException e) {
-                log.error("Failed to set collection exerise to live state - {}", e);
+        String tagString = event.getTag();
+        EventService.Tag tag;
+
+        try {
```
[the remark this answers is not in the document]
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/message/impl/CollectionExerciseEventInboundReceiver.java]: Done

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/message/impl/CollectionExerciseEventInboundReceiver.java [hunk 2]
```diff
@@ -34,19 +34,33 @@
...
+            // If it's a go_live event then attempt to transition the state of the collection exercise to live
+            if (tag == EventService.Tag.go_live){
+                if (messageType == CollectionExerciseEventPublisher.MessageType.EventElapsed) {
+                    UUID collexId = event.getCollectionExerciseId();
+
+                    try {
+                        this.collectionExerciseService.transitionCollectionExercise(collexId,
+                                CollectionExerciseDTO.CollectionExerciseEvent.GO_LIVE);
+                    } catch (CTPException e) {
+                        log.error("Failed to set collection exerise to live state - {}", e);
```
[the remark this answers is not in the document]
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/message/impl/CollectionExerciseEventInboundReceiver.java]: Done

**Comment by madeye-matt** [comment]
@benjefferies I've added tests

**Comment by madeye-matt** [comment]
[Production Go Live dates not transitioning LIVE statuses](https://trello.com/c/mja5AbBP/168-production-go-live-dates-not-transitioning-live-statuses)

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/message/impl/CollectionExerciseEventInboundReceiver.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (benjefferies) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/message/impl/CollectionExerciseEventInboundReceiver.java]: much cleaner

**Pull request merged**